ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/web3j/Web3JClient.java

```diff
@@ -103,15 +109,18 @@ protected void handleError(final Throwable error, final boolean couldBeAuthError
             });
     if (maybeUpdatedTime == timeNow) {
       logExecutionClientError(error, couldBeAuthError);
+      executionClientEventsPublisher.onAvailabilityUpdated(false);
     }
   }
 
   protected void handleSuccess() {
     final long lastErrorTime = lastError.getAndUpdate(x -> NO_ERROR_TIME);
     if (lastErrorTime == STARTUP_LAST_ERROR_TIME) {
       eventLog.executionClientIsOnline();
+      executionClientEventsPublisher.onAvailabilityUpdated(true);
```

Comment:
I'm calling the update inside each branch to avoid publishing events with subsequent successes.